src/BeadingStrategy/InwardDistributedBeadingStrategy.cpp

```diff
@@ -11,37 +11,49 @@ InwardDistributedBeadingStrategy::Beading InwardDistributedBeadingStrategy::comp
     Beading ret;
 
     ret.total_thickness = thickness;
-    if (bead_count > 0)
+    if (bead_count == 1)
     {
-        coord_t to_be_divided = thickness - this->getOptimalThickness(bead_count);
+        ret.bead_widths.emplace_back(thickness);
+        ret.toolpath_locations.emplace_back(thickness / 2);
+        ret.left_over = 0;
+    }
+    else if (bead_count > 1)
+    {
+        const float widen_by = static_cast<float>(thickness) / (optimal_width_outer + optimal_width_inner * (bead_count));
+
+        // Outer wall:
+        const coord_t distributed_width_outer = static_cast<coord_t>(optimal_width_outer * widen_by);
+        ret.bead_widths.emplace_back(distributed_width_outer);
+        ret.toolpath_locations.emplace_back(distributed_width_outer / 2);
+
+        // Inwardly distributed inner walls:
+        const coord_t distributed_width_inner = (thickness - distributed_width_outer) / (bead_count - 1);
+        const coord_t to_be_divided = thickness - (distributed_width_outer + distributed_width_inner * (bead_count - 1));
+
         float total_weight = 0;
-        float middle = static_cast<float>(bead_count - 1) / 2;
+        const float middle = static_cast<float>(bead_count - 2) / 2;
         
         auto getWeight = [middle, this](coord_t bead_idx)
         {
-            float dev_from_middle = bead_idx - middle;
+            float dev_from_middle = (bead_idx - 1) - middle;
             return std::max(0.0f, 1.0f - one_over_distribution_radius_squared * dev_from_middle * dev_from_middle);
         };
         
-        for (coord_t bead_idx = 0; bead_idx < bead_count; bead_idx++)
+        for (coord_t bead_idx = 1; bead_idx < bead_count; bead_idx++)
```

Comment:
Maybe don't use a raw loop here but a std::accumulate,
That would als me that float total_weight can be made const

Reply:
Except that no range is involved, and I'm not (lazily?) making a range out of `bead_idx`s just so I can use `accumulate`. Notice also that we start from 1, which we could rectify with `std::next`, but at this point we're already well past the complexity of a simple, readable for loop ;-)

I had a look through the other algorithms... It's a shame they _all_ work on sequences, really.

Reply:
It sure is I can't wait for Ranges
```c++
auto p = bead | drop(1) | accumulate;
```

Reply:
Oh my gosh... It's like Haskell and pipes from the Unix command line are happily married :-D
Please tell me that this will be lazily evaluated as well?